lib/experimental/interactivity-api/class-wp-directive-processor.php

```diff
@@ -69,6 +69,43 @@ public static function has_root_block() {
 		return isset( self::$root_block );
 	}
 
+	/**
+	 * A string containing the main children of interactive.
+	 *
+	 * @var string
+	 */
+	public static $children_of_interactive_block = null;
```

Comment:
I don't think a single pointer works here because multiple interactive blocks can stack. We need an array.

Reply:
I though the same, and in I had it done [in this commit](https://github.com/WordPress/gutenberg/pull/56302/commits/d7e9608e823313d8dfb06c47d04a49f7f93c272d) with arrays.

But the output has been the same. I will add more nesting to mu blocks environment and also more interactive and non interactive. To check if is enough.

Reply:
Yes, we should be checking with multiple nested pairs of interactive/non-interactive blocks.